prow/cmd/generic-autobumper/prcreator/prcreator.go

```diff
@@ -0,0 +1,712 @@
+)
+
+type fileArrayFlag []string
+
+func (af *fileArrayFlag) String() string {
+	return fmt.Sprint(*af)
+}
+
+func (af *fileArrayFlag) Set(value string) error {
+	for _, e := range strings.Split(value, ",") {
+		fn := strings.TrimSpace(e)
+		info, err := os.Stat(fn)
+		if err != nil {
+			return fmt.Errorf("error getting file info for %q", fn)
+		}
+		if info.IsDir() && !strings.HasSuffix(fn, string(os.PathSeparator)) {
+			fn = fn + string(os.PathSeparator)
+		}
+		*af = append(*af, fn)
+	}
+	return nil
+}
+
+// Options is the options for autobumper operations.
+type Options struct {
+	// The target GitHub org name where the autobump PR will be created. Only required when SkipPullRequest is false.
+	GitHubOrg string `yaml:"gitHubOrg"`
+	// The target GitHub repo name where the autobump PR will be created. Only required when SkipPullRequest is false.
+	GitHubRepo string `yaml:"gitHubRepo"`
+	// The name of the branch in the target GitHub repo on which the autobump PR will be based.  If not specified, will be autodetected via GitHub API.
+	GitHubBaseBranch string `yaml:"gitHubBaseBranch"`
+	// The GitHub username to use. If not specified, uses values from the user associated with the access token.
+	GitHubLogin string `yaml:"gitHubLogin"`
+	// The path to the GitHub token file. Only required when SkipPullRequest is false.
+	GitHubToken string `yaml:"gitHubToken"`
+	// The name to use on the git commit. Only required when GitEmail is specified and SkipPullRequest is false. If not specified, uses values from the user associated with the access token
+	GitName string `yaml:"gitName"`
+	// The email to use on the git commit. Only required when GitName is specified and SkipPullRequest is false. If not specified, uses values from the user associated with the access token.
+	GitEmail string `yaml:"gitEmail"`
+	// The oncall address where we can get the JSON file that stores the current oncall information.
+	OncallAddress string `yaml:"onCallAddress"`
+	// The oncall group that is responsible for reviewing the change, i.e. "test-infra".
+	OncallGroup string `yaml:"onCallGroup"`
+	// Whether to skip creating the pull request for this bump.
+	SkipPullRequest bool `yaml:"skipPullRequest"`
+	// Information needed to do a gerrit bump. Do not include if doing github bump
+	Gerrit *Gerrit `yaml:"gerrit"`
+	// The name used in the address when creating remote. This should be the same name as the fork. If fork does not exist this will be the name of the fork that is created.
+	// If it is not the same as the fork, the robot will change the name of the fork to this. Format will be git@github.com:{GitLogin}/{RemoteName}.git
+	RemoteName string `yaml:"remoteName"`
+	// The name of the branch that will be used when creating the pull request. If unset, defaults to "autobump".
+	HeadBranchName string `yaml:"headBranchName"`
+	// Optional list of labels to add to the bump PR
+	Labels []string `yaml:"labels"`
+	// Dryrun skips PR creation
+	Dryrun bool `yaml:"dryrun"`
```

Comment:
This is a redundant option as we already have SkipPullRequest.

Reply:
Good point, the behavior of Dryrun is slightly different though, it still creates fork, does commit, just not git push or pullrequest creation, would you mind if I change the behavior of `SkipPullRequest`?

Reply:
Yea that is fine. I think this is mostly used for testing anyway